frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/overview/workspace_overview_block_component.dart

```diff
@@ -0,0 +1,415 @@
+    super.configuration,
+  });
+
+  @override
+  BlockComponentWidget build(BlockComponentContext blockComponentContext) {
+    final node = blockComponentContext.node;
+    return WorkspaceOverviewBlockWidget(
+      key: node.key,
+      node: node,
+      configuration: configuration,
+      showActions: showActions(node),
+      actionBuilder: (_, state) => actionBuilder(blockComponentContext, state),
+    );
+  }
+
+  @override
+  bool validate(Node node) => node.children.isEmpty;
+}
+
+class WorkspaceOverviewBlockWidget extends BlockComponentStatefulWidget {
+  const WorkspaceOverviewBlockWidget({
+    super.key,
+    required super.node,
+    super.showActions,
+    super.actionBuilder,
+    super.configuration = const BlockComponentConfiguration(),
+  });
+
+  @override
+  State<BlockComponentStatefulWidget> createState() =>
+      _OverviewBlockWidgetState();
+}
+
+class _OverviewBlockWidgetState extends State<WorkspaceOverviewBlockWidget>
+    with
+        BlockComponentConfigurable,
+        BlockComponentTextDirectionMixin,
+        BlockComponentBackgroundColorMixin,
+        TickerProviderStateMixin {
+  @override
+  BlockComponentConfiguration get configuration => widget.configuration;
+
+  @override
+  EditorState get editorState => context.read<EditorState>();
+
+  @override
+  Node get node => widget.node;
+
+  static const double startingLeftIndent = 50.0;
+
+  static const double leftIndentIncrementValue = 14.0;
+
+  String get viewId => context.read<DocumentBloc>().view.id;
+
+  final ValueNotifier<bool> _isExpandedNotifier = ValueNotifier(true);
+
+  bool get _isExpanded => _isExpandedNotifier.value;
+
+  late final AnimationController _expansionController;
+  late final Animation<double> _curvedAnimationExpansionController;
+  static const _expansionAnimationDuration = Duration(milliseconds: 400);
+
+  @override
+  void initState() {
+    super.initState();
+    _prepareAnimations();
+  }
+
+  void _prepareAnimations() {
+    _expansionController = AnimationController(
+      duration: _expansionAnimationDuration,
+      reverseDuration: _expansionAnimationDuration,
+      vsync: this,
+    );
+
+    _curvedAnimationExpansionController = CurvedAnimation(
+      parent: _expansionController,
+      curve: Curves.fastOutSlowIn,
+      reverseCurve: Curves.fastOutSlowIn,
+    );
+    _runExpandCheck();
+  }
+
+  void _runExpandCheck() {
+    if (_isExpanded) {
+      _expansionController.forward();
+    } else {
+      _expansionController.reverse();
+    }
+  }
+
+  @override
+  void didUpdateWidget(WorkspaceOverviewBlockWidget oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    _runExpandCheck();
+  }
+
+  @override
+  void dispose() {
+    _expansionController.stop();
+    _expansionController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<Either<ViewPB, FlowyError>>(
+      future: ViewBackendService.getAllLevelOfViews(viewId),
```

Comment:
normally, the future value should be fetched only once.

https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html